numismatic/exchanges.py

```diff
@@ -1,31 +1,76 @@
 import logging
-import websockets
 import asyncio
 import json
 import time
+import abc
+from pathlib import Path
+import gzip
+
+from streamz import Stream
 import attr
+import websockets
 
 from .events import Heartbeat, Trade
 
 
 logger = logging.getLogger(__name__)
 
+LIBRARY_NAME = 'numismatic'
+
+
+@attr.s
+class Exchange(abc.ABC):
+    '''Base class for Exchanges'''
+    output_stream = attr.ib()
+    raw_stream = attr.ib(default=None)
+    batch_size = attr.ib(default=1)
+
+    @abc.abstractmethod
+    async def listen(self, symbol):
+        if self.raw_stream is not None:
+            if self.raw_stream is True:
```

Comment:
As Dean pointed out, there is currently no way to hit this code path. Given that the filenames are automatically generated, it would be nice to have a simple options to write the files to the user's cache directory. What's a good way of achieving this?